use-cases/using-Intel-QAT-and-kata.md

```diff
@@ -0,0 +1,484 @@
+Use this file to build a container image with an optimized OpenSSL engine for 
+Intel QAT. Using Docker build with the kata runtime can sometimes have issues. 
+Therefore, it is recommended to change the default runtime to runc before 
+doing a build. Instructions for this are below.
+
+
+> **Note: The Intel QAT driver version in this container might not match the 
+> Intel QAT driver compiled and loaded on the host when compiling.**
+
+```sh
+$ cd $QAT_SRC
+$ curl -O $QAT_DOCKERFILE
+$ sudo sed -i 's/kata-runtime/runc/g' /etc/systemd/system/docker.service.d/50-runtime.conf
+$ sudo systemctl daemon-reload && sudo systemctl restart docker
+$ sudo docker build -t openssl-qat-engine .
+```
+
+### Test Intel QAT in Docker
+
+The host should already be setup with 16 virtual functions of the Intel QAT 
+card bound to VFIO-PCI. This can be verified by looking in `/dev/VFIO` for a 
+listing of functions. Replace the number 90 with one of the VF’s exposed 
+in `/dev/VFIO`. It might be required to add an `IPC_LOCK` capability to your 
+Docker runtime depending on which rootfs is used. 
+
+```sh
+$ sudo docker run -it --runtime=kata-runtime --cap-add=IPC_LOCK --device=/dev/vfio/90 -v /dev:/dev -v ${QAT_CONF_LOCATION}:/etc openssl-qat-engine bash
+```
+
+Below are some commands to run in the container image to verify Intel QAT is 
+working
+
+```sh
+bash-5.0# cat /proc/modules
+bash-5.0# adf_ctl restart
+bash-5.0# adf_ctl status
+bash-5.0# openssl engine -c -t qat
+
+#Test with Intel QAT card acceleration
+bash-5.0# openssl speed -engine qat -elapsed -async_jobs 72 rsa2048 
+
+#Test with CPU acceleration
+bash-5.0# openssl speed -elapsed rsa2048
+```
+
+### Troubleshooting
+
+* Check that `/dev/VFIO` has VF’s enabled
+
+```sh
+$ ls /dev/VFIO
+57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  VFIO
+```
+
+* Check that the modules are loaded when inside the Kata Container
+
+```sh
+bash-5.0# egrep "qat|usdm_drv" /proc/modules
+qat_c62xvf 16384 - - Live 0x0000000000000000 (O)
+usdm_drv 86016 - - Live 0x0000000000000000 (O)
+intel_qat 184320 - - Live 0x0000000000000000 (O)
+```
+
+* Verify that at least the first `c6xxvf_dev0.conf` files is mounted inside the 
+container image in `/etc`. You will need one configuration file for each VF 
+passed into the container.
+
+```sh
+bash-5.0# ls /etc
+c6xxvf_dev0.conf   c6xxvf_dev11.conf  c6xxvf_dev14.conf  c6xxvf_dev3.conf  c6xxvf_dev6.conf  c6xxvf_dev9.conf  resolv.conf
+c6xxvf_dev1.conf   c6xxvf_dev12.conf  c6xxvf_dev15.conf  c6xxvf_dev4.conf  c6xxvf_dev7.conf  hostname
+c6xxvf_dev10.conf  c6xxvf_dev13.conf  c6xxvf_dev2.conf   c6xxvf_dev5.conf c6xxvf_dev8.conf  hosts
+```
+
+* Check `dmesg` inside the container to see if there are any issues with the 
+Intel QAT driver
+
+* If there are issues building the OpenSSL Intel QAT container image then 
```

Comment:
Add in a comma after "image" 

- If there are issues building the OpenSSL Intel QAT container image, then ...

Reply:
Done